### ISSUE TITLE
Training non-English ASR model

### ISSUE THREAD

**Issue by dmitrytyrin** [NVIDIA/NeMo#234]
Hello!
I tried to train russian ASR model based on 1_ASR_tutorial_using_NeMo.ipynb notebook (from NeMo/examples/asr/notebooks/) using Google Colab. I used config jasper_an4.yaml and quartznet5x3.yaml with changed labels accordingly to russian alphabet ("а", "б", "в", "г" and etc.) and wav files with russian speech.
During training I got empty "Reference" in every step. There was "Prediction" on the first step and then only empty rows. Loss seems to be correct but WER is infinite in this case. Is there a problem with the encoding?
I got predictions only with whitespaces on inference, again WER is infinite.

I will appreciate for any hint how to solve this issue.

**Comment by okuchaiev** [comment]
I think the problem is that, by default, AudioToTextDataLayer uses "normalize_transcripts=true" https://nvidia.github.io/NeMo/collections/nemo_asr.html?highlight=audiototextdatalayer#nemo_asr.data_layer.AudioToTextDataLayer   This normalization will remove all non-En characters (it also does some other stuff like converting 25 -> "twenty five").

I recommend you pre-process your data yourself and set this normalization to false like so: https://github.com/NVIDIA/NeMo/blob/master/examples/asr/configs/jasper10x5.yaml#L7  (but set it to false).

Please let us know if this solves your issue

**Comment by dmitrytyrin** [comment]
@okuchaiev, thank you very much! I added row in config file and in data_layer. Now everything works fine.

**Comment by jiamingkong** [comment]
Hi, I tried to train the quartznet model on mandarin data, and would like to load the checkpoint and finetune on it. Reading the training script I realized that the vocabulary file is generated by scanning through AISHELL-2 dataset, which is not openly available. Is there a chance that you could release this vocab file?

**Comment by Slyne** [comment]
@jiamingkong I think there is a vocab file in the checkpoints. https://ngc.nvidia.com/catalog/models/nvidia:aishell2_jasper10x5dr/files

**Comment by jiamingkong** [comment]
@Slyne Thank you so much, this vocab.txt seemed to work!

**Comment by ZerozeroAndre** [comment]
@dmitrytyrin, What dataset did you use?

**Comment by dmitrytyrin** [comment]
@ZerozeroAndre, https://voice.mozilla.org/en/datasets

**Comment by yingxingdechibang** [comment]
@dmitrytyrin hello, I almost have the same problem with you,  empty "Reference" and WER is infinite. I tried to "normalize_transcripts=false", but still didn't work. I wonder what's "I added row in config file and in data_layer. " this mean?

**Comment by dmitrytyrin** [comment]
@yingxingdechibang, hello! I trained model with Google Colab and added rows here:

data_layer_train = nemo_asr.AudioToTextDataLayer(
    manifest_filepath=train_manifest,
    sample_rate=sample_rate,
    labels=labels,
    batch_size=32,
    **normalize_transcripts=False,**
    **params['AudioToTextDataLayer']['train'])
(and in eval datalayer too)

and in config file (jasper_an4.yaml for example):

AudioToTextDataLayer:
    **normalize_transcripts: false**

**Comment by yingxingdechibang** [comment]
@dmitrytyrin @okuchaiev  hello, I tried to use the pretrained model "jasper10x5dr.yaml", "an4 test manifest" as testset ,then run  "jasper_aishell_infer.py" to eval the pretrained model, but I got empty "Reference" and WER is infinite. As you showed above, I set "normalize_transcripts=False" both in collections/nemo_asr/data_layer.py and jasper10x5dr.yaml, but still did't work. Fianlly, I set all  "normalize_transcripts=False" everywhere in the project, still didn't work. I am so confused.

**Comment by wptoux** [comment]
@yingxingdechibang The an4 dataset contains only the lower case english letters, but the vocab file for the aishell-2 datasets contains only the upper case letters. So all the words in the test set will be "normalized", causing an infinite WER.

**Comment by yingxingdechibang** [comment]
@wptoux Thank you so much! It seemed to work!

**Comment by jonathan-cohen-nvidia** [comment]
@okuchaiev Seems like we should add some checks for things like case mismatch to catch errors like this in the future.

**Comment by soheiltehranipour** [comment]
> @yingxingdechibang, hello! I trained model with Google Colab and added rows here:
> 
> data_layer_train = nemo_asr.AudioToTextDataLayer(
> manifest_filepath=train_manifest,
> sample_rate=sample_rate,
> labels=labels,
> batch_size=32,
> **normalize_transcripts=False,**
> **params['AudioToTextDataLayer']['train'])
> (and in eval datalayer too)
> 
> and in config file (jasper_an4.yaml for example):
> 
> AudioToTextDataLayer:
> **normalize_transcripts: false**

Would you please share this Colab file you mentioned?

**Comment by soheiltehranipour** [comment]
> @yingxingdechibang, hello! I trained model with Google Colab and added rows here:
> 
> data_layer_train = nemo_asr.AudioToTextDataLayer(
> manifest_filepath=train_manifest,
> sample_rate=sample_rate,
> labels=labels,
> batch_size=32,
> **normalize_transcripts=False,**
> **params['AudioToTextDataLayer']['train'])
> (and in eval datalayer too)
> 
> and in config file (jasper_an4.yaml for example):
> 
> AudioToTextDataLayer:
> **normalize_transcripts: false**

Is there any chance that i can have the Colab file?

Thanks

**Comment by viveksj** [comment]
I have a favor to ask: To those who have tried this ^, can y'all help answer this: 
[https://github.com/NVIDIA/NeMo/issues/1057](https://github.com/NVIDIA/NeMo/issues/1057)

I followed all instructions on 1_ASR_tutorial_using_NeMo and finished training. 
Now I want to use inference on local files and I'm unsure of the syntax. I also used jasper_an4.yaml

**Comment by okuchaiev** [comment]
closing as this is related to the old version

**Comment by samabdullah** [comment]
do you have nay idea how we will change the config file from command prompt
